x/evm/keeper/call_contract.go

```diff
@@ -63,9 +63,11 @@ func (k Keeper) CallContractWithInput(
 	evmResp, err = k.ApplyEvmMsg(
 		ctx, evmMsg, evmObj, evm.NewNoOpTracer(), commit, txConfig.TxHash, true,
 	)
+	if evmResp != nil {
+		ctx.GasMeter().ConsumeGas(evmResp.GasUsed, "CallContractWithInput")
```

Comment:
We should always consume gas after `ApplyEvmMsg()`, before returning away from `CallContractWithInput()`, or else we lose the opportunity to apply the `EvmResp.GasUsed` variable.